src/modules/integrations/hostctl.nix

```diff
@@ -0,0 +1,44 @@
+{ pkgs, lib, config, ... }:
+
+let
+  hostContent = lib.concatStringsSep "\n" config.hosts;
+  hostHash = builtins.hashString "sha256" hostContent;
+  file = pkgs.writeText "hosts" ''
+    # ${hostHash}
+    ${hostContent}
+  '';
+in
+{
+  options = {
+    hostsProfileName = lib.mkOption {
+      type = lib.types.str;
+      default = "devenv-${builtins.hashString "sha256" config.env.DEVENV_ROOT}";
+      description = "Profile name to use.";
+    };
+
+    hosts = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = "List of hosts entries.";
+    };
+  };
+
+  config = lib.mkIf (builtins.length config.hosts != 0) {
```

Comment:
if `builtins.length config.hosts == 0` then we should run deactivate-hosts if there is `$DEVENV_STATE/hostctl`.

See src/modules/languages/rust.nix how to do the merging to implement this.

Then we don't need to tell how to revert these changes.

Reply:
But the problem is how does the user deactive the hosts when the leaves the shell?